data/pools/transactionPool.go

```diff
@@ -103,12 +104,17 @@ type BlockEvaluator interface {
 	PaySetSize() int
 	TransactionGroup(txads []transactions.SignedTxnWithAD) error
 	Transaction(txn transactions.SignedTxn, ad transactions.ApplyData) error
-	GenerateBlock() (*ledgercore.ValidatedBlock, error)
+	GenerateBlock(addrs []basics.Address) (*ledgercore.UnfinishedBlock, error)
 	ResetTxnBytes()
 }
 
+// VotingAccountChecker provides a list of possible participating account addresses valid for a given round.
+type VotingAccountChecker interface {
+	VotingAccountsForRound(basics.Round) []basics.Address
+}
+
```

Comment:
Why is this a `Checker` rather than `Supplier`?  What does it check?  Can you explain in the comment, or if it doesn't check something, use `Supplier`?